networking/eth2/src/test/java/tech/pegasys/artemis/networking/eth2/compression/SnappyCompressorTest.java

```diff
@@ -0,0 +1,121 @@
+  private final Compressor compressor = new SnappyCompressor();
+
+  @Test
+  public void roundTrip() throws Exception {
+    final BeaconState state = dataStructureUtil.randomBeaconState(0);
+    final Bytes serializedState =
+        Bytes.wrap(SimpleOffsetSerializer.serialize(state).toArrayUnsafe());
+
+    final Bytes compressed = compressor.compress(serializedState);
+    assertThat(compressed).isNotEqualTo(serializedState);
+
+    final Bytes uncompressed = compressor.uncompress(compressed);
+    assertThat(uncompressed).isEqualTo(serializedState);
+  }
+
+  @Test
+  public void uncompress_invalidData() {
+    final BeaconState state = dataStructureUtil.randomBeaconState(0);
+    final Bytes serializedState =
+        Bytes.wrap(SimpleOffsetSerializer.serialize(state).toArrayUnsafe());
+
+    assertThatThrownBy(() -> compressor.uncompress(serializedState))
+        .isInstanceOf(CompressionException.class);
+  }
+
+  @Test
+  public void uncompress_seriesOfValues() throws Exception {
+    final BeaconState stateA = dataStructureUtil.randomBeaconState(0);
+    final BeaconState stateB = dataStructureUtil.randomBeaconState(1);
+    final Bytes serializedStateA =
+        Bytes.wrap(SimpleOffsetSerializer.serialize(stateA).toArrayUnsafe());
+    final Bytes serializedStateB =
+        Bytes.wrap(SimpleOffsetSerializer.serialize(stateB).toArrayUnsafe());
+
+    final Bytes compressedA = compressor.compress(serializedStateA);
+    final Bytes compressedB = compressor.compress(serializedStateB);
+    final Bytes compressedSeries = Bytes.concatenate(compressedA, compressedB);
+    final InputStream input = new ByteArrayInputStream(compressedSeries.toArrayUnsafe());
+
+    // Get first value
+    final Bytes uncompressed = compressor.uncompress(input, serializedStateA.size());
+    assertThat(uncompressed).isEqualTo(serializedStateA);
+    // Then next value
+    final Bytes uncompressed2 = compressor.uncompress(input, serializedStateB.size());
+    assertThat(uncompressed2).isEqualTo(serializedStateB);
+    // Input stream should now be closed
+    assertThat(input.available()).isEqualTo(0);
+    assertThat(input.read()).isEqualTo(-1);
+  }
+
+  @Test
+  public void uncompress_partialValue() throws Exception {
+    final BeaconState state = dataStructureUtil.randomBeaconState(0);
+    final Bytes serializedState =
+        Bytes.wrap(SimpleOffsetSerializer.serialize(state).toArrayUnsafe());
+
+    final Bytes compressed = compressor.compress(serializedState);
+    final int maxBytes = MAX_FRAME_CONTENT_SIZE / 2;
+    // Check assumptions
+    assertThat(serializedState.size()).isGreaterThan(MAX_FRAME_CONTENT_SIZE);
+
+    final InputStream input = new ByteArrayInputStream(compressed.toArrayUnsafe());
+    final Bytes uncompressed = compressor.uncompress(input, maxBytes);
+    assertThat(uncompressed.size()).isLessThanOrEqualTo(maxBytes);
+  }
+
+  @Test
+  @Disabled("SnappyCompressor will currently block in this case")
```

Comment:
We probably want to get this test working before we merge.  Will unfortunately have to be multithreaded I guess and likely not all that easy to write but seems quite important.

Reply:
This test is enabled and works now.